src/frontend/parallelization_phase.nit

```diff
@@ -66,15 +65,38 @@ private class ParallelizationPhase
 			classname += nmethdef.n_methid.as(AIdMethid).n_id.text
 		end
 
-		# Create a string corresponding to the threaded class
-		var s ="""
+		# String corresponding to the generated class
+		var s : nullable String = null
+
+		# case if the method has parameters
+		if nmethdef.n_signature.n_params.length != 0 then
+			var params = ""
+			for param in nmethdef.n_signature.n_params do
+				params += """
```

Comment:
Instead of accumulating strings with `+=` which is so very slow, you should accumulate them in an array and `join` them.